### PR TITLE
FPASF-261: Add CLI commands for admin tasks

### PR DESCRIPTION
**Pull request by MarcUsher**
### Change description
Adds CLI commands for what were previously API endpoints for common admin tasks: retrieving failed or successful submissions and reingesting files (either as batch from s3 storage or a single local file).

CLI commands are now namespaced for clarity: existing database-related CLI commands now require `flask db-data <command_name>` to run, while new admin commands need `flask admin <command_name> <args>`.

- [X] Unit tests and other appropriate tests added or updated
- [X] README and other documentation has been updated / added (if needed)
- [X] Commit messages are meaningful and follow good commit message guidelines (e.g. "FS-XXXX: Add margin to nav items preventing overlapping of logo")

### CODE REVIEW

File: data_store/controllers/admin_tasks.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        except NoResultFound as error:
+            raise error
+
+        if submission:
+            fund_name = (
+                "Pathfinders" if submission.programme_junction.programme_ref.fund.fund_code == "PF" else "Towns Fund"
+            )
+            reporting_round = submission.programme_junction.reporting_round
+            account_id, user_email = submission.submitting_account_id, submission.submitting_user_email
+            db.session.close()  # ingest (specifically `populate_db`) wants to start a new clean session/transaction
```
Comment (samuelhwilliams) [inline data_store/controllers/admin_tasks.py:32]: 🥲  this feels like a bit of a leak/smell here. Sad times. (not your fault)

File: data_store/cli.py [hunk 2]
```diff
@@ -109,5 +111,44 @@ def retrieve_failed(failure_uuid):
...
+        Example usage:
+            flask admin reingest-file <filepath> <submission_id>
+        """
+        with current_app.app_context():
+            print(f"Reingesting submission {submission_id} from {filepath}.")
+            reingest_file(filepath, submission_id)
+
+    @admin_cli.command("reingest-s3")
+    @click.argument("filepath", required=True, type=click.Path(exists=True, dir_okay=False, file_okay=True))
+    def reingest_files_from_s3(filepath):
```
Comment (samuelhwilliams) [inline data_store/cli.py:134]: Nice idea to add options for reingesting from local and from s3 👍

File: data_store/aws.py [hunk 3]
```diff
@@ -85,13 +85,14 @@ def get_file_metadata(bucket_name: str, file_key: str) -> Dict[str, str]:
...
-        response = _S3_CLIENT.head_object(Bucket=bucket_name, Key=file_key)
+        s3_response = _S3_CLIENT.head_object(Bucket=bucket_name, Key=file_key)
     except ClientError as error:
-        raise ClientError(
-            (f"Could not get metadata for {file_key}. "),
-        ) from error
+        if error.response["Error"]["Message"] == "Not Found":
+            raise FileNotFoundError(
+                (f"Could not get metadata for {file_key}. "),
+            ) from error
```
Comment (samuelhwilliams) [inline data_store/aws.py:93]: Should we `raise error` in the case it's not a "Not Found" something? Also - is "Not Found" the code, or a human readable description? If there's a code we can check instead, that's probably safer?

**Comment by samuelhwilliams** [comment]
This all looks really good ✅ 👏 

Ping me for a re-review when it's all ready

**Review by samuelhwilliams — APPROVED** [review]
Looks good 👍  A couple of small nice-to-have tweaks

We'll also need a corresponding update to the docker-runner to change `flask seed-ref` to `flask db-data seed-ref`

File: data_store/cli.py [hunk 4]
```diff
@@ -60,3 +74,80 @@ def drop():
             db.create_all()
 
         print("Database dropped.")
+
+    @admin_cli.command("retrieve-successful")
+    @click.argument("submission_id")
+    def retrieve_successful(submission_id):
+        """CLI command to retrieve a successful submission file from S3.
```
[the remark this answers is not in the document]
Comment (MarcUsher) [reply data_store/cli.py:81]: Yes good shout! Will make that change now. Have got the docker-runner change ready to go as well, when we merge this I'll add it as a new commit on the monolith branch (seems unnecessary to put up a PR for that small change)

**Pull request merged**